app/src/main/java/org/mozilla/focus/widget/DownloadListAdapter.java

```diff
@@ -70,60 +70,83 @@ private void delete(int position){
-        return new DownloadViewHolder(itemView);
+    public int getItemViewType(int position) {
+        return mDownloadInfo.size() >0 ? VIEW_TYPE_NON_EMPTY : VIEW_TYPE_EMPTY;
     }
 
     @Override
-    public void onBindViewHolder(DownloadViewHolder holder, int position) {
-        DownloadInfo downloadInfo = mDownloadInfo.get(position);
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+
+        View itemView;
+        if (VIEW_TYPE_NON_EMPTY == viewType){
+            itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.download_menu_cell,parent,false);
+            return new DownloadViewHolder(itemView);
+        }else {
+            itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.download_empty,parent,false);
+            return new DownloadEmptyViewHolder(itemView);
+        }
+    }
 
-        holder.title.setText(downloadInfo.getFileName());
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
 
-        String subtitle="";
-        if ("successful".equalsIgnoreCase(downloadInfo.getStatus())) {
-            subtitle = downloadInfo.getSize() + "," + downloadInfo.getDate();
-        } else {
-            subtitle = downloadInfo.getStatus();
-        }
+        if (viewHolder instanceof DownloadViewHolder){
+            DownloadViewHolder holder = (DownloadViewHolder) viewHolder;
+            DownloadInfo downloadInfo = mDownloadInfo.get(position);
 
-        holder.subtitle.setText(subtitle);
-
-        holder.action.setTag(position);
-        holder.action.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                final int position = (int) view.getTag();
-                final PopupMenu popupMenu = new PopupMenu(view.getContext(),view);
-                popupMenu.getMenuInflater().inflate(R.menu.menu_delete,popupMenu.getMenu());
-                popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-                    @Override
-                    public boolean onMenuItemClick(MenuItem menuItem) {
-
-                        switch (menuItem.getItemId()){
-                            case R.id.remove:
-                                remove(position);
-                                break;
-                            case R.id.delete:
-                                delete(position);
-                                break;
-                            default:
-                                break;
-                        }
-                        popupMenu.dismiss();
-                        return false;
-                    }
-                });
+            holder.title.setText(downloadInfo.getFileName());
 
-                popupMenu.show();
+            String subtitle="";
+            if ("successful".equalsIgnoreCase(downloadInfo.getStatus())) {
```

Comment:
Try NOT to use plain text "successful" everywhere. At least make it a static final in DownloadInfo